### ISSUE TITLE
Plex with nginx-ingress and Cloudflare is reporting clients using IPv6 as LAN IPs

### ISSUE THREAD

**Issue by onedr0p** [onedr0p/k3s-gitops#12]
I'm seeing some remote users IPs coming in as 10.x.x.x addresses, which doesn't make any sense. I'm not sure if this is an issue with nginx ingress, Cloudflare or what. Very strange issue.

Possible work around: I might just enable Plex relay instead of using nginx. I'll also gain streaming w/ SSL which didn't work with nginx ingress due to lack of a custom certificate set on Plex

**Comment by carpenike** [comment]
Curious to see how you make out on this. The relay sucks, only 2mbps.

**Comment by onedr0p** [comment]
What do you mean 2mbps? I've used the Plex relay in the past before my Kubernetes cluster and my users we're getting comparable speeds.

I'm sure my issue is related to Cloudflare unless your seeing the same issue with LAN IPs.

**Comment by carpenike** [comment]
https://support.plex.tv/articles/216766168-accessing-a-server-through-relay/

> Limitations
> When relaying a connection through us, there are limitations for the streaming qualities you can use:
> Free users are limited to 1 Mbps maximum for streams
> Plex Pass subscribers are limited to 2 Mbps maximum for streams

**Comment by onedr0p** [comment]
I had to read up a little more, it looks like the relay is only used when there is no direct connection to the Plex server, e.g. you don't open the port on your router, or the person streaming has some crazy firewall rules.

This is my understanding because my last iterations of my Plex server definitely had people streaming at a much higher rate than 2mbps and I was using their remote access.

**Comment by onedr0p** [comment]
Send like disabling ipv6 in my Cloudflare account for my domain may be a working if I want to continue using nginx. Sadly this is a issue the Plex developers need to fix....

https://forums.plex.tv/t/plex-api-sessions-status-not-honoring-x-forwarded-for-x-real-ip-with-ipv6-clients/175705/36

**Comment by onedr0p** [comment]
> At Cloudflare we believe in being good to the Internet and good to our customers. By moving on from the legacy world of IPv4-only to the modern-day world where IPv4 and IPv6 are treated equally, we believe we are doing exactly that. In the Cloudflare dashboard, IPv6 is no longer something you can toggle on and off, it’s always just on. However, if you need to turn off IPv6 (and say goodbye to the modern internet), you may still do so using the Cloudflare API.

**Comment by onedr0p** [comment]
I am going to enable remote access in Plex and enable the relay, I will keep this issue open so others can see into this.

**Comment by onedr0p** [comment]
Opened port on my router to point to my Plex LoadBalancer IP and enabled Remote Access.

**Comment by onedr0p** [comment]
Well fuck, looks like this doesn't work either: I am streaming from my lan and getting a 10.x.x.x address treated as WAN traffic.

![image](https://user-images.githubusercontent.com/213795/74461947-4e408e00-4e5d-11ea-850b-60c7f159e8d7.png)

**Comment by onedr0p** [comment]
Switched back to nginx w/ Plex, I'll be posting debug logs once I see my issue again but I doubts there's much I can do.

**Comment by onedr0p** [comment]
Logs indicate that one of the users with a local IP is connecting via IPv6

```
Feb 17, 2020 21:00:53.338 [0x7f33f9796700] VERBOSE - X-Forwarded-For: 2604:6000:110d:424b:ac98:ffb7:e2ac:393f
Feb 17, 2020 21:00:53.338 [0x7f33f9796700] VERBOSE - Auth: We found auth token (xxxxxxxxxxxxxxxxxxxx), enabling token-based authentication.
Feb 17, 2020 21:00:53.338 [0x7f33f9796700] DEBUG - Auth: authenticated user 2618380 as redacted@gmail.com
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] DEBUG - Request: [10.42.4.156:48542 (Subnet)] GET /photo/:/transcode?width=397&height=596&minSize=1&upscale=1&url=%2Flibrary%2Fmetadata%2F32598%2Fthumb%2F1579557972%3FX-Plex-Token%3Dxxxxxxxxxxxxxxxxxxxx (23 live) GZIP Signed-in Token (redacted@gmail.com)
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * accept => */*
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * Accept-Encoding => gzip
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * accept-language => en-US,en;q=0.9
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * CDN-Loop => cloudflare
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * CF-Connecting-IP => 2604:6000:110d:424b:ac98:ffb7:e2ac:393f
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * CF-IPCountry => US
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * CF-RAY => 566c68f548445580-ORD
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * CF-Visitor => {"scheme":"https"}
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * Host => plex.redacted.io
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * origin => http://app.plex.tv
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * referer => http://app.plex.tv/
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * sec-fetch-mode => cors
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * sec-fetch-site => cross-site
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * user-agent => Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Forwarded-For => 2604:6000:110d:424b:ac98:ffb7:e2ac:393f
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Forwarded-Host => plex.redacted.io
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Forwarded-Port => 443
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Forwarded-Proto => https
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Original-Forwarded-For => 2604:6000:110d:424b:ac98:ffb7:e2ac:393f
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Plex-Token => xxxxxxxxxxxxxxxxxxxx
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Real-IP => 2604:6000:110d:424b:ac98:ffb7:e2ac:393f
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Request-ID => 165aadb0b633dd1e02df7b0b634c599e
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * X-Scheme => https
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * width => 397
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * height => 596
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * minSize => 1
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * upscale => 1
Feb 17, 2020 21:00:53.338 [0x7f33f0ff9700] VERBOSE -  * url => /library/metadata/32598/thumb/1579557972?X-Plex-Token=xxxxxxxxxxxxxxxxxxxx
```

**Comment by onedr0p** [comment]
Changing that setting in Cloudflare (`Network -> Pseudo IPv4 -> Overwrite Headers`) has _fixed_ my issue for now. This is still an issue though and I'll keep this issue open until Plex has fixed it.

**Comment by onedr0p** [comment]
I haven't had an issue after making the changes above, I will report back if Plex ever fixes IPv6 so I can disable that setting in Cloudflare.

**Comment by onedr0p** [comment]
Looks like Plex fixed this with v1.19.1.2589

I'll be able to disable that setting in Cloudflare!
